src/pypdfium2/_helpers/pageobjects.py

```diff
@@ -155,6 +157,121 @@ def transform(self, matrix):
             raise PdfiumError("Failed to transform pageobject with matrix.")
 
 
+class PdfTextObj (PdfObject):
+    """
+    Textobject helper class.
+    
+    You may want to call :meth:`.PdfPage.get_objects` or :meth:`.PdfTextPage.get_textobj` to obtain an instance of this class.
+    """
+    
+    def __init__(self, *args, textpage=None, **kwargs):
+        
+        # If there is a textpage, assign `page` and `pdf`.
+        # `page` being non-None will set needs_free to False
+        # If the object does not belong to a page, it will be freed.
+        if textpage is not None:
+            kwargs.update(page=textpage.page, pdf=textpage.page.pdf)
+        
+        super().__init__(*args, **kwargs)
+        self.textpage = textpage
+        
+        # If there is a textpage, the object should not be freed, but we attach a parent holder finalizer
+        if textpage is not None:
+            self._close_func = PdfTextObj._holder_impl
+            self._ex_args = (textpage, )
+            self._attach_finalizer()
```

Comment:
Hmm, I don't think this would work with `PdfPage.remove_obj()`, in the `textpage is not None` case.
`remove_obj()` would need to activate a passive `FPDFPageObj_Destroy()` finalizer, but this doesn't work when we already have an active holder finalizer instead...

Also, `remove_obj()` says any `PdfTextPage` handles to the page ought to be closed before removing a text object. So perhaps it is really the page that needs to stay alive, rather than the textpage?
I need to re-read the API docs...

Reply:
Huh, `FPDFText_GetTextObject()` merely says
> The caller does not own the returned object.

That doesn't make clear whether the object is owned by the textpage, or the page.